indico/web/client/js/react/components/IButton.tsx

```diff
@@ -0,0 +1,79 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import React from 'react';
+
+import {toClasses} from 'indico/react/util';
+
+interface IButtonPropTypes {
```

Comment:
this can't be "inlined"? :/

Reply:
It can, but not sure if it makes it better:

```ts
export default function IButton({
  classes = {},
  href,
  title,
  children,
  onClick,
  highlight = false,
  disabled = false,
  icon = '',
  dropdown = false,
  borderless = false,
  style = {},
}: {
  classes: object;
  href: string | undefined;
  title: string;
  children: React.ReactNode;
  onClick: (() => undefined) | undefined;
  highlight: boolean;
  disabled: boolean;
  icon: string;
  dropdown: boolean;
  borderless: boolean;
  style: object;
}) {
 ...
}
```

Reply:
Yeah, I find setting a type more user-friendly and not that difference from what we were doing with PropTypes already.

Reply:
fine for me, I thought maybe something similar to python's type annotations exited there, ie having the type next to the name of the prop

Reply:
Can you annotate dict keys in Python, though? I think the closest thing we have is TypedDicts, right?

Reply:
true, even though I'd say the dict keys here are closer to kwargs than to dict keys...